hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/db/Proto2Codec.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.apache.hadoop.hdds.utils.db;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.MessageLite;
+import com.google.protobuf.Parser;
+import org.apache.ratis.util.function.CheckedFunction;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.IntFunction;
+
+/**
+ * Codecs to serialize/deserialize Protobuf v2 messages.
+ */
+public final class Proto2Codec<M extends MessageLite>
+    implements Codec<M> {
+  private static final ConcurrentMap<Class<?>, Codec<?>> CACHE
+      = new ConcurrentHashMap<>();
+
+  /** @return the {@link Codec} for the given class. */
+  public static <T extends MessageLite> Codec<T> get(Class<T> clazz) {
+    final Codec<?> codec = CACHE.computeIfAbsent(
+        clazz, c -> new Proto2Codec<>(clazz));
+    return (Codec<T>) codec;
+  }
+
+  private static <M extends MessageLite> Parser<M> getParser(Class<M> clazz) {
```

Comment:
Nit: `M` here is unrelated to the instance's `M` type, so it might be better to define as `T`, similar to `get`'s type.  (Same applies to `Proto3Codec`, sorry I didn't notice it previously.)

Reply:
Sure.